core/CORE_NAILS_Lang.php

```diff
@@ -11,6 +11,11 @@
  * @link
  */
 
+if (!class_exists('MX_Lang')) {
+
+    require NAILS_COMMON_PATH . 'MX/Lang.php';
```

Comment:
From memory, I think `MX_Lang` also requires various other classes to be loaded... maybe not - again, you probably implemented this like so for a reason.

Reply:
@hellopablo it's got hard dependencies on `\CI_Lang`, which it extends, and `\CI` and `\Modules`, which it references in its `::load` method.

I _think_ we could do a similar check in `core/CORE_NAILS_Lang.php` and `require` the files with the class definitions for `CI_Lang` and `Modules` as per above, but including the `\CI` class (provided by MX) will automatically invoke CodeIgniter's super-controller routine, which I'm guessing will break everything.

At any rate, I only added this manual loading of `MX_Lang` to enable `CORE_NAILS_Lang` to extend from it and thus load successfully. 

`MX_Lang` extends from CodeIgniter's `CI_Lang`, which _does_ have a `::__construct` method, which calls what I think is a CI function called `log_error`.

This function must be getting defined somewhere during the CLI runtime stack, otherwise this inherited `__construct` method would cause a fatal error. However, it would be easy to add a `__construct` override method to `\CORE_NAILS_Lang` which would do a similar check for the existence of the `\CI` class — if it exists, call `parent::__construct`, if not, do nothing.

It doesn't seem like this is necessary, but this is dealing with quite a convoluted mesh of hard-dependencies, so unless you think adding the `__construct` override might have unintended consequences, shall I go ahead and add it?

Reply:
Eep. I'm not sure. There's loads of scenarios to test for. I'd be inclined to not edit these files directly, but perhaps (as part of the `console.php` setup) define empty classes which do the absolute minimum we need in order for the rest to work, i.e:

```
class CI {}
class MX_Lang {
   function something() {
      return null
   }
}
```

Comprende? As you said, it's a big ol' mesh of dependencies - I've tried to tackle this many times before and never managed.

Have a think, and a test, but it might be one we tackle together on one machine.